v22.2/fips-compliance.md

```diff
@@ -0,0 +1,162 @@
+---
+title: Install a FIPS-compliant build of CockroachDB
+summary: Learn how to ensure that CockroachDB runs in an environment that enforces compliance with FIPS 140-2.
+toc: true
+docs_area: deploy
+---
+
+## Overview of FIPS 140-2 compliance in CockroachDB
+
+{{site.data.alerts.callout_info}}
+FIPS-compliant builds of CockroachDB are **experimental** and are not yet qualified for production use.
+{{site.data.alerts.end}}
+
+[Federal Information Processing Standards (FIPS) 140-2](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.140-2.pdf) is a mandatory standard for the protection of sensitive or valuable data within systems used by and for United States and Canadian federal government agencies. In the United States, the standard is applicable to all federal agencies that use cryptographic-based security systems to protect sensitive information in computer and telecommunication systems (including voice systems) as defined in Section 5131 of the Information Technology Management Reform Act of 1996, Public Law 104-106 and the Federal Information Security Management Act of 2002, Public Law 107-347.
+
+When a cryptographic module or library has a FIPS 140-2 certificate, it has been tested and formally validated by the U.S. and Canadian Governments.
+
+The standard CockroachDB binaries rely on the Go language for cryptographic functions and do not use OpenSSL libraries installed in the operating system. Go's cryptographic libraries are not FIPS-validated.
+
+FIPS-compliant CockroachDB binaries are built using [Red Hat's Go FIPS with OpenSSL toolchain](https://github.com/golang-fips/go), which contains the necessary modifications for the Go crypto library to use an external cryptographic library in a FIPS compliant way. FIPS-compliant CockroachDB runtimes use the host operating system's OpenSSL libraries and commands, rather than Go's cryptographic libraries. When the installed OpenSSL has a FIPS 140-2 certificate and [FIPS mode is enabled in the Linux kernel](#enable-fips-mode-in-the-linux-kernel), the CockroachDB runtime is compliant with FIPS 140-2. FIPS-compliant binaries and Docker images are available for CockroachDB v22.2.7 and above. CockroachDB FIPS-compliant runtimes run only on Intel 64-bit Linux systems.
```

Comment:
This entire paragraph is grasping at broken straws and fails to make the most important point: _we need to communicate that we "pass through" any and all guarantees about FIPS by delegating our usage of cryptography to someone else_ (subtext: if the customer wants guarantees/certifications they can go and talk to that other party and not us)

To make that point, it confuses matters to talk about technology in detail (especially so if the technology details are wrong, see comments above). Instead, we must emphasize the architectural idea: "passing through the responsibility". Something along the following lines:

"""
Cockroach Labs offers FIPS compliance in CockroachDB as follows:
- a special build of CockroachDB (hereafter referred to as "FIPS-enabled") was modified to  delegate uses of cryptography for most production features to the OS-provided OpenSSL library.
- the customer is expected to deploy the FIPS-enabled build on top of a FIPS compliant deployment of the OS and the OpenSSL library.
- the customer is expected to audit their application and validate themselves that they only use CockroachDB features for which cryptography passes through to OpenSSL, as per the table below.

Only when the FIPS-enabled CockroachDB build is deployed on top of a FIPS-validated platform and OpenSSL, and the application is guaranteed to only use features for which cryptography is served in CockroachDB through OpenSSL, can the CockroachDB deployment be considered compliant with FIPS standard.
"""